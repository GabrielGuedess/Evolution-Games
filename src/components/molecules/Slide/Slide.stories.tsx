import { Story, Meta } from '@storybook/react/types-6-0';

import { Slide, SlideProps } from './Slide';

export default {
  title: 'Molecules/Slide',
  component: Slide,
  args: {
    srcSlideImage: '/img/games/theLastOfUs.png',
    altSlideImage: 'The Last Of Us',
    srcHighlightImage: '/img/games/highlightName/nameTheLastOfUs.png',
    altHighlightImage: 'The Last Of Us Highlight',
  },
} as Meta<SlideProps>;

export const Default: Story<SlideProps> = args => (
  <div style={{ width: '100%', height: '100vh' }}>
    <Slide {...args} />
  </div>
);
