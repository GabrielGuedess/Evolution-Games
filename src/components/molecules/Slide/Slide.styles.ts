import media from 'styled-media-query';

import { Container } from 'components/atoms/Container/Container';

import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  height: 100%;
  height: 100%;
`;

export const WrapperSlide = styled.div`
  ${({ theme }) => css`
    position: relative;
    height: 100%;
    height: 100%;

    img {
      mask-image: linear-gradient(
        180deg,
        transparent 4%,
        ${theme.colors.white} 56%,
        transparent 100%
      );
    }
  `}
`;

export const HighlightName = styled(Container)`
  ${({ theme }) => css`
    position: absolute;
    inset: 0;
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-end;
    padding-top: 12rem;
    z-index: ${theme.layers.base};

    ${media.greaterThan('medium')`
      padding-top: 20rem;
    `}
  `}
`;
