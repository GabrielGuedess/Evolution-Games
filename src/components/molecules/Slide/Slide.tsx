import Image from 'next/image';

import * as S from './Slide.styles';

import 'swiper/css';
import 'swiper/css/effect-fade';
import 'swiper/css/pagination';

export type SlideProps = {
  srcSlideImage: string;
  altSlideImage: string;
  srcHighlightImage: string;
  altHighlightImage: string;
};

export const Slide = ({
  srcSlideImage,
  altSlideImage,
  srcHighlightImage,
  altHighlightImage,
}: SlideProps) => (
  <S.Wrapper>
    <S.WrapperSlide>
      <Image
        src={srcSlideImage}
        layout="fill"
        objectFit="cover"
        quality={100}
        alt={altSlideImage}
      />
    </S.WrapperSlide>
    <S.HighlightName>
      <Image
        src={srcHighlightImage}
        width={768}
        height={296}
        objectFit="contain"
        quality={100}
        alt={altHighlightImage}
        data-swiper-parallax="-300"
      />
    </S.HighlightName>
  </S.Wrapper>
);
