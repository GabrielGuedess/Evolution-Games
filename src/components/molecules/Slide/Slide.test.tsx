import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { Slide, SlideProps } from './Slide';

const props: SlideProps = {
  srcSlideImage: '/img/games/theLastOfUs.jpg',
  altSlideImage: 'The Last Of Us',
  srcHighlightImage: '/img/games/highlightName/nameTheLastOfUs.png',
  altHighlightImage: 'The Last Of Us Highlight',
};

describe('<Slide />', () => {
  it('should render the Image background and highlight name', () => {
    // Arrange
    const { container } = renderWithTheme(<Slide {...props} />);

    // Assert
    expect(
      screen.getByRole('img', { name: props.altSlideImage }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('img', { name: props.altHighlightImage }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
