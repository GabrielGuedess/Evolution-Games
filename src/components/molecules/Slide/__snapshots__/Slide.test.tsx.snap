// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Slide /> should render the Image background and highlight name 1'] = `
.c2 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

.c0 {
  height: 100%;
  height: 100%;
}

.c1 {
  position: relative;
  height: 100%;
  height: 100%;
}

.c1 img {
  -webkit-mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
  mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
}

.c3 {
  position: absolute;
  inset: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding-top: 12rem;
  z-index: 10;
}

@media (min-width:1170px) {
  .c2 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c2 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c2 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width:768px) {
  .c3 {
    padding-top: 20rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <img
        alt="The Last Of Us"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
      <noscript />
    </span>
  </div>
  <div
    class="c2 c3"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27768%27%20height=%27296%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="The Last Of Us Highlight"
        data-nimg="intrinsic"
        data-swiper-parallax="-300"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
  </div>
</div>
`;
